src/Compilers/VisualBasic/Portable/SymbolDisplay/CustomSymbolDisplayFormatter.vb

```diff
@@ -124,5 +120,23 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         Public Shared Function ShortNameWithTypeArgsAndContainingTypes(symbol As Symbol) As FormattedSymbol
             Return New FormattedSymbol(symbol, ShortWithTypeArgsAndContainingTypesFormat)
         End Function
+
+        Public Shared Function DefaultErrorFormat(symbol As Symbol) As FormattedSymbol
+            Return New FormattedSymbol(symbol, SymbolDisplayFormat.VisualBasicErrorMessageFormat)
+        End Function
+
+        Public Shared Function DefaultErrorFormatIfErrorSymbol(arg As Object) As Object
+
+            Dim symbol = TryCast(arg, Symbol)
+            If symbol Is Nothing Then
+                Return arg
+            End If
+
+            If TypeOf symbol Is ITypeSymbol AndAlso DirectCast(symbol, ITypeSymbol).SpecialType <> SpecialType.None Then
```

Comment:
In compiler we prefer Kind checks to type checks. Also, compilers do not operate in terms of symbol interfaces.